### PR TITLE
build: create unified libOS+enclave bundle

### PR DESCRIPTION
**Pull request by mythi**
Fixes: #212 
Fixes: #128

### CODE REVIEW

File: tools/packaging/build/build_payload.sh [hunk 1]
```diff
@@ -22,16 +22,12 @@ export PAYLOAD_ARTIFACTS="${SCRIPT_ROOT}/payload_artifacts"
 mkdir -p ${PAYLOAD_ARTIFACTS}
 
 # build pre-installed OCI bundle for agent enclave container
-pushd ${SCRIPT_ROOT}/agent-enclave-bundle
-docker build ${ENCLAVE_CC_ROOT} -f ${SCRIPT_ROOT}/agent-enclave-bundle/Dockerfile --build-arg SGX_MODE=${SGX_MODE} --build-arg KBC=${KBC} -t agent-instance
+pushd ${SCRIPT_ROOT}/unified-bundle
+docker build ${ENCLAVE_CC_ROOT} -f ${SCRIPT_ROOT}/unified-bundle/Dockerfile --build-arg SGX_MODE=${SGX_MODE} --build-arg KBC=${KBC} -t unified-instance
 jq -a -f sgx-mode-config.filter config.json.template | tee ${PAYLOAD_ARTIFACTS}/config.json
-docker export $(docker create agent-instance) | tee > ${PAYLOAD_ARTIFACTS}/agent-instance.tar
+docker export $(docker create unified-instance) | tee > ${PAYLOAD_ARTIFACTS}/unified-instance.tar
```
[the remark this answers is not in the document]
Comment (mythi) [reply tools/packaging/build/build_payload.sh:28]: `>` pipes `stdout` from `tee` to that file. Without it, we'd see all the binary garbage printed to `stdout` and piped to the file. The alternative would be `tee 1>/dev/null ${..}/unified-instance.tar`

**Comment by mythi** [comment]
I believe we could get this bigger restructuring/build optimization merged.

**Comment by mythi** [comment]
@dcmiddle @piotrpalcz this PR implements significant (IMO) simplifications to the enclave-cc stack. It now unifies the two "bundles" into one. In addition, my POC to unify the agent-enclave and app-enclave functionality into a single enclave also seems to work. This would allow us to experiment with sealing policies based on MRenclave.

**Comment by dcmiddle** [comment]
Non blocking question... in tools/packaging/build/unified-bundle/ I can't tell what the original intent was of enclave-agent-cc-kbc.yaml nor enclave-agent-sample-kbc.yaml
They are identical files and I can't find references to them elsewhere in coco.

**Review by dcmiddle — APPROVED** [review]

**Comment by mythi** [comment]
> I can't tell what the original intent was of enclave-agent-cc-kbc.yaml nor enclave-agent-sample-kbc.yaml

we used to have a third copy for eaa-kbc that was different. that was removed but the `KBC` build arg was kept.

**Comment by piotrpalcz** [comment]
LGTM, tested and working

**Pull request merged**